### PR TITLE
Modify relu native implementation 2

### PR DESCRIPTION
**Pull request by wojtuss**
### PR types
Bug fixes

### PR changes
OPs

### Describe
This is an alternative to https://github.com/PaddlePaddle/Paddle/pull/30980 fix for https://github.com/PaddlePaddle/Paddle/issues/30905. It separates the CPU and GPU `relu` kernels, keeping the original implementation for GPU.

### CODE REVIEW

File: cmake/cuda.cmake [hunk 1]
```diff
@@ -216,6 +216,8 @@ endif(WIN32)
 set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -w")
 # Set :expt-relaxed-constexpr to suppress Eigen warnings
 set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --expt-relaxed-constexpr")
+# Set :expt-extended-lambda to enable HOSTDEVICE annotation on lambdas
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --expt-extended-lambda")
```
Comment (arogowie-intel) [inline cmake/cuda.cmake:220]: Now this flag is unnecessary.

Comment (wojtuss) [reply cmake/cuda.cmake:220]: I thought so too, but the error message when building with CUDA says something different:
```
2021-02-12 17:57:55 /paddle/build/third_party/eigen3/src/extern_eigen3/unsupported/Eigen/CXX11/src/Tensor/TensorExecutor.h(610): error: The closure type for a lambda ("lambda [](float)->float", defined at /paddle/paddle/fluid/operators/activation_op.h:325) cannot be used in the template argument type of a __global__ function template instantiation, unless the lambda is defined within a __device__ or __global__ function, or the lambda is an 'extended lambda' and the flag --expt-extended-lambda is specified
```

**Comment by wojtuss** [comment]
@luotao1 ,
In the coverage report of the failed PR-CI-Coverage build it looks like the coverage was measured on an optimized (Release) build, and the missing lines were optimized out by inlining. Could you please help resolve the building issue?

**Comment by wojtuss** [comment]
@luotao1 ,
Could you please also suggest how to fix the PR-CI-OP-benchmark build which complains about missing test script for `elu`? This PR does not touch that functor.

**Comment by lidanqing-vv** [comment]
> @luotao1 ,
> Could you please also suggest how to fix the PR-CI-OP-benchmark build which complains about missing test script for `elu`? This PR does not touch that functor.

Juncai replied at https://github.com/PaddlePaddle/Paddle/pull/30980#issuecomment-781018287

**Comment by juncaipeng** [comment]
> > @luotao1 ,
> > Could you please also suggest how to fix the PR-CI-OP-benchmark build which complains about missing test script for `elu`? This PR does not touch that functor.
> 
> Juncai replied at [#30980 (comment)](https://github.com/PaddlePaddle/Paddle/pull/30980#issuecomment-781018287)

Once @luotao1  approves this PR, the OP-benchmark error can be fixed.

**Review by luotao1 — APPROVED** [review]
LGTM

**Review by juncaipeng — APPROVED** [review]
LGTM

**Pull request merged**